controllers/mytba_controller.py

```diff
@@ -19,21 +21,38 @@ def get(self):
         self._require_registration('/account/register')
 
         user = self.user_bundle.account.key
+        now = datetime.datetime.now()
         team_favorites_future = Favorite.query(Favorite.model_type == ModelType.TEAM, ancestor=user).fetch_async()
 
         live_events = EventHelper.getWeekEvents()
+        all_events_this_year_future = EventListQuery(now.year).fetch_async()
         favorite_team_keys = map(lambda f: ndb.Key(Team, f.model_key), team_favorites_future.get_result())
         favorite_teams_future = ndb.get_multi_async(favorite_team_keys)
 
+        future_events = filter(lambda e: e.start_date > now, all_events_this_year_future.get_result())
+        future_eventteams_futures = []
+        for event in future_events:
+            future_eventteams_futures.append(EventTeamsQuery(event.key_name).fetch_async())
```

Comment:
I'm going to say do the TeamYearEvents query on all favorites, then filter.

I think the average user has less than 50 favorites, and filtering each team's events for the next one is less work than filtering all events' event teams for all favorite teams.